edb/server/compiler/compiler.py

```diff
@@ -387,6 +387,8 @@ class CompilerState:
 
     @functools.cached_property
     def state_serializer_factory(self) -> sertypes.StateSerializerFactory:
+        # TODO: This factory will probably need to become per-db once
```

Comment:
@msullivan, need your input here.  Config spec can already be different w.r.t enabled extensions, or am I confusing things here?

Reply:
We don't support doing session configs with extension configs yet, so I think it is fine

Reply:
Hopefully I will implement that in the next month but I'll try to fix it then